plugin/helpers_bots.go

```diff
@@ -45,7 +114,10 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 	// If the bot has already been created, there is nothing to do.
 	if botIdBytes != nil {
 		botId := string(botIdBytes)
-		return botId, nil
+		setImagesErr := p.setBotIconAndProfileImage(botId, o.ProfileImagePath, o.IconImagePath)
+		if setImagesErr != nil {
+			return botId, setImagesErr
```

Comment:
shouldn't this be `return "", err`?

Reply:
@ali-farooq0 It depends if we still want to do something with the id if the bot is successfully created, but setting of the image fails. I was thinking it should still be returned, but happy to remove.

Reply:
Ah true, however the caller would not know the type of the error unfortunately, since we only have that information in the text. I would recommend to just return `"", err` in this error case.